src/osd/OSDMap.cc

```diff
@@ -5216,6 +5223,19 @@ void OSDMap::rm_all_upmap_prims(CephContext *cct, OSDMap::Incremental *pending_i
   }
 }
 
+void OSDMap::rm_all_upmap_prims(CephContext *cct, OSDMap::Incremental *pending_inc) {
+  vector<pg_t> upmap_prims;
+  get_upmap_primary_pgs(&upmap_prims);
+  for (auto pg : upmap_prims) {
```

Comment:
Wouldn't it be simpler to just iterate over `pg_upmap_primaries` and using the structured bindings instead of creating the temporary?

```cpp
for (const auto& [pg, _] : pg_upmap_primaries) {
  //
```

Reply:
ACK. I removed the supporting function and took this simpler route.